src/silo/query_engine/filter_expressions/nucleotide_symbol_equals.cpp

```diff
@@ -28,18 +27,20 @@ std::unique_ptr<silo::query_engine::operators::Operator> NucleotideSymbolEquals:
    const silo::DatabasePartition& database_partition,
    Expression::AmbiguityMode mode
 ) const {
-   if (position >= database.reference_genome->genome_segments[0].size()) {
+   const auto& seq_store_partition =
+      database_partition.nuc_sequences.at(database.database_config.default_nucleotide_sequence);
```

Comment:
Same here: This probably also needs to be generalized at some point?

Reply:
Yes, many expressions need to support multiple segment targets. This will be done when we also support amino acids

Reply:
... as I want to decouple changes to storage and queries where it is not absolutely necessary. Therefore the PR for AAcids will also first only cover storage and then a second PR will cover queries (including the namings here)